bytecode/instr.py

```diff
@@ -268,11 +269,17 @@ def lineno(self):
     def lineno(self, lineno):
         self._set(self._name, self._arg, lineno)
 
+    @property
+    def stack_effect(self):
+        # All opcodes whose arguments are not represented by integers have
+        # a stack_effect indepent of their argument.
+        arg = (self._arg if isinstance(self._arg, int) else
+               0 if self._opcode >= _opcode.HAVE_ARGUMENT else None)
```

Comment:
Sorry, but I dislike the idea of Instr.stack_effect(): for a jump using a label, the size of the instruction depends on the distance between the jump and the label.

That's why I proposed to only add stack_effect() to ConcreteInstr.

The problem is that you also want to use a CFG which uses abstract bytecode (and so labels) :-p Maybe bytecode API should be extended to support CFG with concrete bytecode?

My design issue: if jump argument is an integer, I would prefer to have an immutable bytecode object.